components/protocol-wasm/world.wit

```diff
@@ -0,0 +1,87 @@
+/// Exposes functions to perform package validations and inclusion proofs from the `protocol` and `transparency` crates
+/// Protobuf structures for validations can be found in the warg.proto file
+/// Protobuf structures for inclusion proofs can be found in the proof.proto file
+
+default world component {
+  export protocol: self.protocol
+}                
+
+interface protocol {
+    record proto-envelope-body {
+      content-bytes: list<u8>,
+      key-id: string,
+      signature: string
+    }
+    enum hash-algorithm {
+      sha256
+    }
+    record dyn-hash {
+      algo: hash-algorithm,
+      bytes: list<u8>
+    }
+    variant record-id {
+      dyn-hash(dyn-hash)
+    }
+    record head {
+      digest: record-id,
+      timestamp: option<string>
+    }
+    enum permission {
+      release,
+      yank
+    }
+    record permission-entry {
+      key-id: string,
+      permissions: list<permission>
+    }
+    record key-entry {
+      key-id: string,
+      public-key: string
+    }
+    record released {
+      content: dyn-hash
+    }
+    record yanked {
+      by: string,
+      timestamp: string
+    }
+    variant release-state {
+      released(released),
+      yanked(yanked)
+    }
+    record release {
+      version: string,
+      by: string,
+      timestamp: string,
+      state: release-state
+    }
+    
+    record validator {
+      algorithm: option<hash-algorithm>,
+      head: option<head>,
+      permissions: list<permission-entry>,
+      releases: list<release>,
+      keys: option<list<key-entry>>,
+    }
+    record log-leaf {
+      log-id: string,
+      record-id: string
+    }
+    record package-info {
+      name: string,
+      checkpoint: option<string>,
+      state: validator,
+      heads: list<log-leaf>
+    }
+    record map-checkpoint {
+      log-root: string,
+      log-length: u32,
+      map-root: string
+    }
+    record inclusion {
+      log: string,
+      map: string
+    }
+    validate: func(package-records: list<proto-envelope-body>) -> package-info
```

Comment:
Turn validator into a "psuedo-resource" that can be initialized empty or with JSON starting contents and has methods that let you move the validator forward one package record at a time.

See wasi-streams [`input-stream`](https://github.com/WebAssembly/wasi-io/blob/main/wit/streams.wit#L30) for an example of a pseudo-resource.